packages/ses/test/test-scope.js

```diff
@@ -0,0 +1,341 @@
+  const originalFoobar = globalLexicals.foobar;
+  doAssignment('foobar', foobar);
+  t.is(globalLexicals.foobar, foobar);
+  t.not(globalObject.foo, originalFoobar);
+  t.is(globalObject.foobar, undefined);
+  t.is(globalThis.foobar, undefined);
+
+  t.is(Object.keys(globalObject).length, 3);
+  t.is(Object.keys(globalLexicals).length, 1);
+
+  delete globalThis.bar;
+});
+
+test('scope behavior - strict vs sloppy locally non-existing global set', t => {
+  t.plan(4);
+
+  const globalObject = {};
+  const { safeEvaluate: evaluateStrict } = makeSafeEvaluator({
+    globalObject,
+    sloppyGlobalsMode: false,
+  });
+  const { safeEvaluate: evaluateSloppy } = makeSafeEvaluator({
+    globalObject,
+    sloppyGlobalsMode: true,
+  });
+
+  globalThis.bar = {};
+
+  // Known compromise in fidelity of the emulated script environment:
+  t.notThrows(() => evaluateStrict('bar = 123'));
+  t.throws(() => evaluateStrict('abc = 123'), {
+    instanceOf: ReferenceError,
+  });
+  t.notThrows(() => evaluateSloppy('bar = 456'));
+  t.notThrows(() => evaluateSloppy('xyz = 456'));
+
+  delete globalThis.bar;
+});
+
+test('scope behavior - realm globalThis property info leak', t => {
+  t.plan(8);
+
+  const globalObject = {};
+  const { safeEvaluate: evaluate } = makeSafeEvaluator({
+    globalObject,
+  });
+
+  t.is(evaluate('typeof foo'), 'undefined');
+  t.is(evaluate('typeof bar'), 'undefined');
+  t.throws(() => evaluate('foo'), { instanceOf: ReferenceError });
+  t.throws(() => evaluate('bar'), { instanceOf: ReferenceError });
+
+  globalThis.bar = {};
+
+  t.is(evaluate('typeof foo'), 'undefined');
+  t.is(evaluate('typeof bar'), 'undefined');
+  t.throws(() => evaluate('foo'), { instanceOf: ReferenceError });
+  // Known compromise in fidelity of the emulated script environment:
+  t.is(evaluate('bar'), undefined);
+
+  delete globalThis.bar;
+});
+
+test('scope behavior - Symbol.unscopables fidelity test', t => {
+  t.plan(12);
+
+  const globalObject = {
+    Symbol,
+    [Symbol.unscopables]: { bar: true },
+  };
+  const { safeEvaluate: evaluate } = makeSafeEvaluator({
+    globalObject,
+  });
+
+  t.is(evaluate('typeof foo'), 'undefined');
+  t.is(evaluate('typeof bar'), 'undefined');
+  t.throws(() => evaluate('foo'), { instanceOf: ReferenceError });
+  t.throws(() => evaluate('bar'), { instanceOf: ReferenceError });
+
+  globalThis.bar = {};
+
+  t.is(evaluate('typeof foo'), 'undefined');
+  t.is(evaluate('typeof bar'), 'undefined');
+  t.throws(() => evaluate('foo'), { instanceOf: ReferenceError });
+  // Known compromise in fidelity of the emulated script environment:
+  t.is(evaluate('bar'), undefined);
+
+  evaluate('this[Symbol.unscopables] = { bar: true }');
+
+  t.is(evaluate('typeof foo'), 'undefined');
+  t.is(evaluate('typeof bar'), 'undefined');
+  t.throws(() => evaluate('foo'), { instanceOf: ReferenceError });
+  // Known compromise in fidelity of the emulated script environment:
+  t.is(evaluate('bar'), undefined);
```

Comment:
I think the overload of a `bar` test is confusing here. I'd expect the aberration to be:

```js
evaluate('this.baz = {}; this[Symbol.unscopables] = { baz: true }; ');
// Known compromise in fidelity of the emulated script environment:
t.throws(() => evaluate('baz'), { instanceOf: ReferenceError });
```

Reply:
yes, the aberration is very different in the current version vs the triple backflip refactor. this test was mostly added to show off the aberration the backflip introduces.